modules/cudaoptflow/misc/python/test/test_nvidiaopticalflow.py

```diff
@@ -0,0 +1,33 @@
+import os
+import cv2 as cv
+import numpy as np
+
+from tests_common import NewOpenCVTests, unittest
+
+class nvidiaopticalflow_test(NewOpenCVTests):
+    def setUp(self):
+        super(nvidiaopticalflow_test, self).setUp()
+        if not cv.cuda.getCudaEnabledDeviceCount():
+            self.skipTest("No CUDA-capable device is detected")
+
+    @unittest.skipIf('OPENCV_TEST_DATA_PATH' not in os.environ,
+                        "OPENCV_TEST_DATA_PATH is not defined")
+    def test_calc(self):
+        frame1 = os.environ['OPENCV_TEST_DATA_PATH'] + '/gpu/opticalflow/frame0.png'
+        frame2 = os.environ['OPENCV_TEST_DATA_PATH'] + '/gpu/opticalflow/frame1.png'
+
+        npMat1 = cv.cvtColor(cv.imread(frame1),cv.COLOR_BGR2GRAY)
+        npMat2 = cv.cvtColor(cv.imread(frame2),cv.COLOR_BGR2GRAY)
+
+        cuMat1 = cv.cuda_GpuMat(npMat1)
+        cuMat2 = cv.cuda_GpuMat(npMat2)
+
+        nvof = cv.cuda_NvidiaOpticalFlow_1_0.create(cuMat1.shape[1], cuMat1.shape[0], 5, False, False, False, 0)
```

Comment:
Thank you for Python test!

Please add handling for non-supported CUDA devices.
Similar to `SkipTestException` logic in C++ tests.

Reply:
> Please add handling for non-supported CUDA devices.
> Similar to `SkipTestException` logic in C++ tests.

Somthing like this? I am not familiar with python interface. But test could run only if compute version of GPU is 7.5 and above.

        device_prop = cv.cuda.cudaGetDeviceProperties(None, 0)
        if device_prop.major < 7 and device_prop.minor < 5:
            self.skipTest("No Supported CUDA-capable device is detected")

Reply:
Use `try...except` (see try-catch code in C++ tests):

```
except cv.error as e:
    if e.code == cv.Error.StsBadFunc or e.code == cv.Error.StsBadArg or e.code == cv.Error.StsNullPtr:
        return  # algorithm is not supported/implemented in the current environment
```

Reply:
Great. Thanks :-)